BraveShared/PaymentRequest/PaymentRequestBody.swift

```diff
@@ -0,0 +1,51 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+struct PaymentRequestBody: Decodable {
+    struct PaymentRequestMethodData: Decodable {
+        struct SupportedInstrumentsData: Decodable {
+            let supportedNetworks: [String]
+            let supportedTypes: [String]?
+        }
+        
+        let supportedMethods: String
+        let data: SupportedInstrumentsData?
+    }
+    
+    struct PaymentRequestDetails: Decodable {
+        struct Item: Decodable {
+            let label: String
+            let amount: Amount
+            
+            struct Amount: Decodable {
+                let currency: String
+                let value: String
+            }
+        }
+        let total: Item
+        let displayItems: [Item]
+    }
+    
+    // name of the message passed from JS to Swift
+    let name: String
```

Comment:
This can probably be omitted entirely. Do we use this at all anywhere?

Reply:
We do: https://github.com/brave/brave-ios/pull/2327/files#diff-4dc70f536f408cb9c86ed24f1c0394eeR55